pkg/deployer/helm/ensure.go

```diff
@@ -106,18 +109,56 @@ func (h *Helm) ApplyFiles(ctx context.Context, files map[string]string, exports
 		return err
 	}
 
-	h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
 	h.DeployItem.Status.ProviderStatus = statusData
-	h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
 	if err := h.kubeClient.Status().Update(ctx, h.DeployItem); err != nil {
 		h.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(h.DeployItem.Status.LastError,
 			currOp, "UpdateStatus", err.Error())
 		return err
 	}
-	h.DeployItem.Status.LastError = nil
+
 	return nil
 }
 
+func (h *Helm) CheckResourcesHealth(ctx context.Context) error {
+	var (
+		currOp      = "CheckResourcesHealthHelm"
+		helmDecoder = serializer.NewCodecFactory(Helmscheme).UniversalDecoder()
+		status      = &helm.ProviderStatus{}
+	)
+
+	if _, _, err := helmDecoder.Decode(h.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
+		return err
+	}
+
+	if len(status.ManagedResources) == 0 {
+		return nil
+	}
+
+	objects := make([]*unstructured.Unstructured, len(status.ManagedResources))
+	for i, ref := range status.ManagedResources {
+		obj := kutil.ObjectFromTypedObjectReference(&ref)
+		objects[i] = obj
+	}
+
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   0,
+		Steps:    3,
+	}
```

Comment:
we should make the max timeout configurable.

Reply:
This is done now in the provider configuration.